### PR TITLE
camel-quarkus-management: Added tests for managed beans

### LINKED ISSUE #4661: camel-quarkus-management: Improve test coverage
Only the CamelContext MBean is tested currently. Add tests for the other MBean classes:
- Component
- Consumer
- DataFormat
- Endpoint
- Processor
- Route
- Service

Also add a test for custom MBean components.

### PR DESCRIPTION
**Pull request by djcoleman**
Fixes #4661

- Added MBeans tests for the built-in types (components, endpoints, services etc)
- Added MBeans test for custom bean
- Updated docs.

[x] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[x] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[x] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[x] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[x] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[x] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests-jvm/management/src/test/java/org/apache/camel/quarkus/component/management/it/ManagementTest.java [hunk 1]
```diff
@@ -24,23 +24,69 @@
...
-    public void testCamelManagement() throws Exception {
-        MBeanServer server = ManagementFactory.getPlatformMBeanServer();
-        ObjectName objectName = new ObjectName("org.apache.camel:type=context,*");
+    public void testManagedBean() throws Exception {
+        ObjectName objectName = new ObjectName("org.apache.camel:type=processors,name=\"counter\",*");
         Set<ObjectInstance> mbeans = server.queryMBeans(objectName, null);
         assertEquals(1, mbeans.size());
-
         ObjectInstance instance = mbeans.iterator().next();
-        String routeXML = (String) server.invoke(instance.getObjectName(), "dumpRoutesAsXml", new Object[] {}, new String[] {});
```
[the remark this answers is not in the document]
Comment (djcoleman) [reply integration-tests-jvm/management/src/test/java/org/apache/camel/quarkus/component/management/it/ManagementTest.java:43]: Sure, I can add that check back in.

[the remark this answers is not in the document]
Comment (djcoleman) [reply integration-tests-jvm/management/src/test/java/org/apache/camel/quarkus/component/management/it/ManagementTest.java:43]: I added that test back in.

**Comment by djcoleman** [comment]
Please can someone merge this as I don't have permission. Thanks.

File: docs/modules/ROOT/pages/reference/extensions/management.adoc [hunk 2]
```diff
@@ -33,6 +33,20 @@ ifeval::[{doc-show-user-guide-link} == true]
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
 endif::[]
 
+[id="extensions-management-usage"]
+== Usage
+For information on using Managed Beans in Camel, consult the https://camel.apache.org/manual/jmx.html[JMX section of the Camel Manual].
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/reference/extensions/management.adoc:38]: Please replace this with an `xref` link. You can find examples in some of the other extension docs. Just grep for `xref:manual`.

**Comment by jamesnetherton** [comment]
> Please can someone merge this as I don't have permission. Thanks.

I left one final nitpick comment. We should be good to merge after fixing that up.

File: docs/modules/ROOT/pages/reference/extensions/management.adoc (hunk 2 above)
Comment (djcoleman) [reply docs/modules/ROOT/pages/reference/extensions/management.adoc]: Done. I think I got the syntax right...

**Pull request merged**